modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/ActionsComponentPropsTransformer.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const ACTIONS = {
+	showInfo() {
+		showSidebar();
+	}
+};
+
+const getComputedWidth = element => {
+	const computedStyle = window.getComputedStyle(element, null);
+
+	const computedWidth = parseFloat(computedStyle.width.replace("px","")) -
+		(parseFloat(computedStyle.paddingLeft.replace("px","")) * 2) -
+		(parseFloat(computedStyle.paddingRight.replace("px","")) * 2);
+
+	return computedWidth;
+};
+
+const showSidebar = () => {
+	const sidebarContainer = document.querySelector('.sidebar-container');
+	const dashboardContents = document.querySelectorAll('.dashboard-content');
+	
+	if(sidebarContainer.classList.contains('in')) {
+		dashboardContents.forEach(element => {
+			element.style.width = '100%';
+	  	});
+  
+	  	sidebarContainer.classList.remove('in');
+	  }
+	  else {
+		const windowWidth = window.outerWidth;
+		const sidebarContainerWidth = sidebarContainer.offsetWidth;
+
+		dashboardContents.forEach(element => {
+			const dashboardContentContainerWidth = 
+				getComputedWidth(element.parentElement);
+		
+			const newDashboardContentContainerWidth = 
+				dashboardContentContainerWidth - 
+				(sidebarContainerWidth -
+					((windowWidth - dashboardContentContainerWidth) / 2)
+				);
+		
+			element.style.width = `${dashboardContentContainerWidth}px`;
+			element.offsetWidth = element.offsetWidth;
+			element.style.width = `${newDashboardContentContainerWidth}px`;
+		});
+
+		sidebarContainer.classList.add('in');
+	}
+};
+
+export default function propsTransformer({
+	items,
+	...otherProps
+}) {
+	return {
+		...otherProps,
+		items: items.map((item) => {
+			return {
+				...item,
+				onClick(event) {
+					const action = item.data?.action;
+
+					if (action) {
+						event.preventDefault();
+
+						ACTIONS[action]();
+					}
+				},
+			};
+		}),
+	};
```

Comment:
As I was saying in the original branch... I'd expect us to keep all of this in React and avoid rendering the sidebar markup in a jsp. That makes it all harder to wire and to reuse as a pattern.

I know we don't have a pattern for this yet, so now might be the time (but don't tell team tango) to explore this a little bit! 😉 

What I was thinking was that we could render the react component from here and then simply have the actions change its state. You would render `<Sidebar />` in a `closed` state and then depending on the action you'd be opening the panel and fetching data and even changing which component renders inside the sidebar panel to show the fetched information :)

@wincent, @brunobasto, @matuzalemsteles, @markocikos... thoughts on this?

Reply:
I thought about that too but the concept of rendering a component from a props transformer seemed weird to me

Reply:
It is, but we never quite get the right APIs at the first try... took us several weeks and attempts to get to `propsTransformer`, so try something and show it and let's see where that takes us.

Reply:
![image](https://media.giphy.com/media/l1AsyvXH44nETtw52/giphy.gif)

Reply:
Can we resolve this? I think so...